File: src/main/java/plus/maa/backend/repository/entity/gamedata/ArkTilePos.java
package plus.maa.backend.repository.entity.gamedata;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Data;

import java.util.List;

/**
 * 地图格子数据
 *
 * @author dragove
 * created on 2022/12/23
 */
@Data
@JsonNaming(PropertyNamingStrategies.LowerCamelCaseStrategy.class)
public class ArkTilePos {

    private String code;
    private Integer height;
    private Integer width;
    private String levelId;
    private String name;
    private String stageId;
    private List<List<Tile>> tiles;
    private List<List<Double>> view;

    @Data
    @JsonNaming(PropertyNamingStrategies.LowerCamelCaseStrategy.class)
    public static class Tile {
        private String tileKey;
        private Integer heightType;
        private Integer buildableType;
        private Boolean isStart;
        private Boolean isEnd;
    }
}

File: src/main/kotlin/plus/maa/backend/repository/entity/gamedata/ArkTilePos.kt
package plus.maa.backend.repository.entity.gamedata

import com.fasterxml.jackson.databind.PropertyNamingStrategies.LowerCamelCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming

/**
 * 地图格子数据
 *
 * @author dragove
 * created on 2022/12/23
 */
@JsonNaming(LowerCamelCaseStrategy::class)
data class ArkTilePos(
    val code: String? = null,
    val height: Int = 0,
    val width: Int = 0,
    val levelId: String? = null,
    val name: String? = null,
    val stageId: String? = null,
    val tiles: List<List<Tile>>? = null,
    val view: List<List<Double>>? = null,
) {
    @JsonNaming(LowerCamelCaseStrategy::class)
    data class Tile(
        val tileKey: String? = null,
        val heightType: Int? = null,
        val buildableType: Int? = null,
        val isStart: Boolean? = null,
        val isEnd: Boolean? = null,
    )
}
